src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp

```diff
@@ -334,4 +322,28 @@ double GuideRate::get_grvalue_result(const GRValState& gr) const
         : std::max(gr.curr.value, 0.0);
 }
 
+
+void GuideRate::checkTimeToUpdate(const double sim_time, const size_t report_step)
+{
+    const auto& config = this->schedule[report_step].guide_rate();
+
+    if (!config.has_model()) {
+        this->time_to_update = false;
+        return;
+    }
+
+    // getting the last update time
+    double last_update_time = std::numeric_limits<double>::max();
+    for ([[maybe_unused]] const auto& [wgname, value] : this->values) {
+        const double update_time = value->curr.sim_time;
+//         if (value->curr.value > 0 && update_time < last_update_time) {
+        if (update_time < last_update_time) {
+            last_update_time = update_time;
+        }
+    }
+
+    const double update_delay = config.model().update_delay();
+    this->time_to_update = (sim_time >= (last_update_time + update_delay));
```

Comment:
```C++
this->guide_rates_expired = !(sim_time < previous_update_time + update_delay);
```

Reply:
I have to say without using `!` is actually more straightforward to me. It is totally sensible if we use a different name, then the following can be better. 

```c++
this->guide_rates_valid = (sim_time < previous_update_time + update_delay);
```